lib/new.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
+const {Color} = require('./color');
+const {HttpRequest} = require('./http_request');
+
+const fs = require('fs');
+const util = require('util');
+const {URL} = require('url');
+const EventEmitter = require('events');
+const fsReadFile = util.promisify(fs.readFile);
+
+let testMode = false;
+
+class App extends EventEmitter {
```

Comment:
format: style mismatch:
```javascript
34|
35|class App extends EventEmitter {
36|  /**
```
```suggestion
class App extends EventEmitter {/**
```

⏎␣⁺␣⁺ at column 33 should be removed.
Triggered rule # 43
```
  +2.roles not in {RIGHT, MAP}
	∧ +2.reserved not in {await}
	∧ ^1.roles not in {CALL, EXPRESSION}
	∧ -1.roles in {EXPRESSION} and not in {STRING}
	∧ -1.reserved not in {:}
	∧ -3.length ≤ 4
	∧ -4.reserved = )
	⇒ y = ∅
	Confidence: 0.767. Support: 15.
```